core/src/main/java/org/elasticsearch/script/ScriptSettings.java

```diff
@@ -141,6 +139,15 @@ public ScriptSettings(ScriptEngineRegistry scriptEngineRegistry, ScriptContextRe
                     scriptModeSettings.add(setting);
                 }
             }
+        };
+        for (Class<? extends ScriptEngineService> c : scriptEngineRegistry.getRegisteredScriptEngineServices()) {
+            if (c != NativeScriptEngineService.class) {
+                String language = scriptEngineRegistry.getLanguage(c);
+                populate.accept(language, scriptEngineRegistry.getDefaultInlineScriptEnableds().get(language));
+            }
+        }
+        if (templateBackend != null) {
```

Comment:
`ScriptSettings` doesn't require it to make testing marginally more convenient. I can change it if you think it is worth it.